skimage/_shared/testing.py

```diff
@@ -361,3 +362,51 @@ def inner(*args, **kwargs):
         return inner
 
     return wrapper
+
+
+def assert_stacklevel(*warnings, offset=-1):
+    """Assert correct stacklevel on a captured warnings.
+
+    When scikit-image raises warnings the stacklevel should usually be set in
+    such a way that the origin of the warning will point to the public function
```

Comment:
```suggestion
    When scikit-image raises warnings, the stacklevel should ideally be set
    so that the origin of the warnings will point to the public function
```